unit_test/blas/Test_Blas_Newton.hpp

```diff
@@ -0,0 +1,197 @@
+#include <gtest/gtest.h>
+
+#include <KokkosBlas_Newton_impl.hpp>
+#include <KokkosKernels_TestUtils.hpp>
+
+namespace Test {
+
+// Logistic equation
+// dy/dt=y(1-y)
+//
+// solution y = 1/(1+exp(-t))
+// y(0)=0.5
+//
+// Using BDF1 to integrate:
+// y-y_n=dt*y*(1-y)
+//
+// Residual: r = y - y_n - dt*y*(1-y)
+// Jacobian: J = 1 - dt + 2*dt*y
+template <typename scalar_type, typename execution_space>
+struct LogisticEquation {
+  using vec_type = Kokkos::View<scalar_type*, execution_space>;
+  using mat_type = Kokkos::View<scalar_type**, execution_space>;
+
+  const int neqs = 1;
+  scalar_type dt;
+  vec_type state;
+
+  LogisticEquation(const scalar_type dt_, vec_type initial_state)
+      : dt(dt_), state(initial_state) {}
+
+  KOKKOS_FUNCTION void derivatives(const scalar_type /*t*/, const vec_type& y,
+                                   const vec_type& dydt) const {
+    dydt(0) = y(0) - state(0) - dt * y(0) * (1 - y(0));
+  }
+
+  KOKKOS_FUNCTION void jacobian(const scalar_type /*t*/, const vec_type& y,
+                                const mat_type& jac) const {
+    jac(0, 0) = 1 - dt + 2 * dt * y(0);
+  }
+
+  KOKKOS_FUNCTION void update_state(const vec_type /*y*/) const {
+    state(0) = state(0);
```

Comment:
Is this a typo?

Reply:
Not really, it used to make sense in the context of the ODE solver but not really for the Newton one. I will just remove this method.